frontend/app/utils/route-utils.ts

```diff
@@ -0,0 +1,12 @@
+import { useMatches } from '@remix-run/react';
+
+import { z } from 'zod';
+
+const pageTitleDataSchema = z.object({ pageTitle: z.string() });
+
+export function usePageTitle() {
+  return useMatches()
+    .map(({ data }) => pageTitleDataSchema.safeParse(data))
+    .map((result) => (result.success ? result.data.pageTitle : undefined))
+    .reduce((last, curr) => curr ?? last, undefined);
```

Comment:
If you set `ES2023` instead of `ES2022` in tsconfig.json lib option you will gain access to [findLast](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast)
```suggestion
  return useMatches().findLast(({ data }) => {
    const result = pageTitleDataSchema.safeParse(data);
    return result.success ? result.data.pageTitle : undefined;
```

Reply:
We can do this in a future PR and change all reducers to `.findLast()` in one commit.

`/wontfix`

Reply:
Alright.